arai60/20-valid-parentheses/step3.py

```diff
@@ -0,0 +1,13 @@
+class Solution:
+    def isValid(self, s: str) -> bool:
+        stack = ["*"]
+        bracket_pairs = {"(": ")", "{": "}", "[": "]", "*": ""}
+
+        for char in s:
+            if char in bracket_pairs:
+                stack.append(char)
+                continue
+            if char != bracket_pairs[stack[-1]]:
```

Comment:
個人的にbracket_pairs[stack[-1]]が少し認知負荷が高いと感じました。
変数の命名により明示的にすると緩和できると思いました。
```suggestion
            expected_bracket = bracket_pairs[stack[-1]]
            if char != expected_bracket:
```

Reply:
ありがとうございます。
確かに改めて見ると認知負荷が高いですね、、「stack の一番後ろにあるということは opening bracket を期待しているのだな」という推察をしてもらわないといけないですもんね。
以下で説明変数を追加しました！
https://github.com/kazizi55/coding-challenges/pull/6/files#diff-dee0cfbbf6f296838839ded17654fe00273d500efa9a5fdb004e516117cecc86R1-R16